[CS2113T-T10-3] Mod Happy

Ch40gRv1-Mu opened the pull request.

ngys117 committed:
Update confirmation to use TextUi instead of Scanner

Ch40gRv1-Mu committed:
Update the Regex for OptionParser

Update the Regex for OptionParser

ngys117 committed:
Update Class Diagram for DeleteCommand

ngys117 committed:
Update Command Diagram Note

chooyikai committed:
Update UG phrasing and formatting

Ch40gRv1-Mu committed:
Fix Coding standards

Yzkkk committed:
added javadoc

heekit73098 committed:
Merge pull request #114 from Yzkkk/master

added junit code for commands

heekit73098 committed:
Update OptionCommand regex and refactor configuration values

chooyikai committed:
Edit UG based on peer review feedback

ngys117 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp into branch-PR-CleanUp

# Conflicts:
#	src/test/java/seedu/duke/parsers/ModHappyParserTest.java

ngys117 committed:
Change TextUi Methods

Update CommandClassDiagram

Include Style.puml in Class Diagrams

heekit73098 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp into branch-expandExceptions

heekit73098 committed:
Fixed regex for Help Command

heekit73098 committed:
Update code quality

ngys117 committed:
Update DG to show abstract and interfaces explicitly

heekit73098 committed:
Fixed JUnit Tests

heekit73098 committed:
Merge pull request #115 from ngys117/branch-PR-CleanUp

Update DG, refactor some code and directories

heekit73098 committed:
Added Content page to DG

ngys117 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp

heekit73098 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp into branch-expandExceptions

heekit73098 committed:
Fixed regex for tag

Yzkkk committed:
minor edit

chooyikai committed:
Merge branch 'master' into minor-edits

heekit73098 committed:
Added Grade and Edit feature, User stories and front matters

chooyikai committed:
Fix class diagrams (hopefully)

Ch40gRv1-Mu committed:
Add TaskDuration class and corresponding JUnit

Add TaskDuration class and corresponding JUnit

Ch40gRv1-Mu committed:
Fix ModHappyParser

Fix ModHappyParser

Ch40gRv1-Mu committed:
Fix coding standard

Fix coding standard

Ch40gRv1-Mu committed:
Update TaskDuration

Update TaskDuration: adding three comments

Ch40gRv1-Mu committed:
Update TextUiTest

Update TextUiTest

Ch40gRv1-Mu committed:
Merge branch 'master' into branch-update-ugdg

Ch40gRv1-Mu committed:
Update UG for duration

Update UG for duration

Ch40gRv1-Mu committed:
Merge branch 'branch-update-ugdg' of https://github.com/Ch40gRv1-Mu/tp into branch-update-ugdg

Ch40gRv1-Mu committed:
Update StringConstants

Update StringConstants on a emoji

Ch40gRv1-Mu committed:
Update TaskDuration

Add upper bound for duration number.

heekit73098 committed:
Merge pull request #124 from Ch40gRv1-Mu/branch-refactor-duration

Add TaskDuration

Ch40gRv1-Mu committed:
Update DG

Update DG

heekit73098 committed:
Merge pull request #118 from Ch40gRv1-Mu/branch-update-ugdg

Branch update ugdg

chooyikai committed:
Merge branch 'master' into minor-edits

heekit73098 committed:
Update DG

chooyikai committed:
Fix #125, update some JUnit tests

heekit73098 committed:
Fix bug and limit the available MCs

chooyikai committed:
Update UG, fix some bugs

heekit73098 committed:
Merge pull request #116 from chooyikai/minor-edits

Minor edits to program help text and UG/DG

heekit73098 committed:
Added Seq Diagrams

heekit73098 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp into branch-expandExceptions

heekit73098 committed:
Update Code Quality

heekit73098 committed:
Fix Code Quality

heekit73098 committed:
Fixed Code Quality

chooyikai committed:
Merge pull request #121 from heekit73098/branch-expandExceptions

Expand Exceptions and updated the JUnit test cases for them

heekit73098 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp into branch-updateDG

heekit73098 committed:
Removed image

heekit73098 committed:
Remove Debug

heekit73098 committed:
Removed debug statement

chooyikai committed:
Merge pull request #126 from heekit73098/branch-expandExceptions

Remove Debug

heekit73098 committed:
Update contents in UG

Ch40gRv1-Mu committed:
Merge pull request #127 from heekit73098/branch-updateDG

Updated DG with front matters, Grade and Edit feature

thedarie commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+> `NoArgumentParser` is an exception to the above; instead of being associated with a single command type, it is responsible for handling all commands which do not accept any arguments.
+
+The following details how the `Parser` component works at runtime:
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ParserSequenceDiagram.puml)
```
![image](https://user-images.githubusercontent.com/74888935/160957543-918d6e1b-c04a-4c69-93f1-ff1a4ee0c790.png)

Should the highlighted portion be in the alt section? Since I don't think it will run after the [else] section

hearobe replied:
Adding on to this comment - I'm not sure what the right way of doing it is, but perhaps you could use an opt block conditioned on [commandWord != XYZ_COMMAND_WORD] that throws ParseException, instead of the current alt block. This way you can keep the happy path in the main body of the diagram which might be clearer!

ngys117 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp

zhengster commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+# Mod Happy: Developer Guide
 
-## Acknowledgements
+## Contents
```
Nice addition of table-of-contents - this was very helpful for navigation! :)

zhengster commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+> `NoArgumentParser` is an exception to the above; instead of being associated with a single command type, it is responsible for handling all commands which do not accept any arguments.
+
+The following details how the `Parser` component works at runtime:
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ParserSequenceDiagram.puml)
```
This diagram is a bit complicated to follow; perhaps you could consider breaking it into smaller fragments, or using reference frames to omit less important details. 
![image](https://user-images.githubusercontent.com/20442226/160970988-2f051548-e43a-4a35-9605-bd4565ce710f.png)

YitHien replied:
Since you have mentioned the exception to be thrown above, is it possible for the same exception to be thrown here as well? (i.e. targetModule of wrong format, missing fields). 
![image](https://user-images.githubusercontent.com/69446722/161180885-d65cbb25-87cc-4012-b8cd-4063e244d12d.png)

zhengster commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/Storage.puml)
+
+Storage is an interface supporting write/read data to/from computer storage:
```
For readability purposes, perhaps it would be better to enclose all mentions of classes or methods in backticks i.e. (`Storage`), which is what you did in previous sections.

zhengster commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
-# Developer Guide
```
In general, a well-formatted and thorough DG - nice job!

zhengster commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+
+## 6. Implementation
+
+This section describes some details on how some features are implemented.
```
A bit redundant given that the title of the section, "Implementation," already implies this - perhaps you could provide more detailed explanation, like "This section provides additional detail on how some core features, such as _____, are implemented."

Yzkkk committed:
minor edits

Yzkkk committed:
resolved merge conflicts

YitHien commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+
+Storage is an interface supporting write/read data to/from computer storage:
+* Storage interface is implemented by JsonStorage in Mod Happy, which will read and load data to and from json format.
+* ListStorage can save a ArrayList of any class that extends Object in json format, and read them back into corresponding objects. (E.g. ModuleListStorage, TaskListStorage inherit from ListStorage)
```
Perhaps its a good idea to be consistent and surround classes and methods in backticks (`ListStorage`, `Storage`)

striris reviewed (commented):
![image](https://user-images.githubusercontent.com/72813713/161181796-59c4b8c9-7213-4923-9284-4c7a520d1171.png)
the sequence diagram in the edit section is not so easy to read because it is too small. consider make it larger?

YitHien commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+The following sequence diagram illustrates the process:
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/EditSeqDiagrams/Edit.puml)
+
```
Consider breaking down the sequence diagram into smaller subsections; certain details are too small to be read easily.
![image](https://user-images.githubusercontent.com/69446722/161182031-c4f5beaa-7c3a-4490-b4c5-65566ef029e2.png)

striris commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+
+### 5.2. Parser Component
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/Parser.puml)
```
A good use of notes in the diagram.

OzairHasan commented on docs/ClassDiagrams/CommandClassDiagram.puml:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    class DeleteCommand {
+    -moduleCode: String
+    -taskModule: String
+    -taskIndex: int
+    +DeleteCommand(moduleCode: String)
+    +DeleteCommand(taskModule: String, index: int)
+    +execute(): CommandResult
+    +getUserConfirmation(): String {static}
+    }
+
+    class TextUi {
+    +getUserCommand(): String {static}
+    +showMessage(): void {static}
+    }
+
+    class ExitCommand {
+    +execute(): CommandResult
+    }
+
+    class CommandResult {
+    +CommandResult(result: String)
+    +toString(): String
+    }
+}
+
+note top of command
+Note:
+This class diagram only illustrates two
+example command types for simplicity.
+end note
+
+note left of DeleteCommand
+Note:
+Dependency on TextUi is
+unique to DeleteCommand.
+end note
+
+@enduml
```
The use of diagrams is appropriate and they are very clear and easy to follow.

YitHien commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+
+Here is an example of editing the description of a task (First task of the module CS2113T):
+
+1. User inputs `edit task 1 -m CS2113T -d "Changed"`.
+2. `ModHappyParser` identifies the command word as `edit` and passes `task 1 -m CS2113T -d "Changed"` to `EditParser`.
+3. `EditParser` instantiates a `EditCommand` with `taskModule = "CS2113T"`, `taskIndex = 0`, `description = "Changed"`, `workingTime = null`, `taskname = null`. This is returned to `Main`.
+4. `Main` calls the `execute()` method of the `EditCommand` instance.
+5. `EditCommand` first gets the relevant `Module` and invokes `editTaskFromModule(targetModule)`.
+6. `editTaskFromModule(targetModule)` retrieves the task `targetTask` specified by the index and invokes `targetTask.setTaskDescription(description)` to change the description.
```
Consider using object diagrams for each step to help readers better visualize the process (same applies for subsequent sections of DG). You may refer to this part of AddressBook's DG below 
![image](https://user-images.githubusercontent.com/69446722/161182988-503b933b-5629-46ce-99ae-2d31657ffaee.png)

striris commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+
+The `Storage` component is responsible for the saving and loading of program data from and to its data files. The following class diagram illustrates the structure of this component:
+
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/ClassDiagrams/Storage.puml)
```
The abstract method is used and the diagram is very clear.

YitHien commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+
+## 10. Instructions for manual testing
 
 {Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
```
Consider commenting/removing away any notes to self

striris commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+
+<br>
+
+### 6.2. Tag Feature
```
This is a very interesting feature! good job!

OzairHasan commented on docs/SequenceDiagrams/Grade.puml:
```diff
@@ -0,0 +1,52 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+
+skinparam shadowing false
+participant ":ModHappyParser" as ModHappyParser
+participant ":GradeParser" as GradeParser
+participant ":GradeCommand" as GradeCommand
+participant ":ModuleList" as ModuleList
+participant ":Module" as Module
+hide footbox
+
+note right of ModHappyParser
+Some methods are omitted from this diagram.
+end note
+
+[->ModHappyParser:parseCommand(userInput)
+activate ModHappyParser
+create GradeParser
+ModHappyParser -> GradeParser: GradeParser()
+activate GradeParser
+return
+
+ModHappyParser -> GradeParser: parseCommand(arguments)
+activate GradeParser
+create GradeCommand
+GradeParser -> GradeCommand: GradeCommand(moduleCode, moduleGrade)
+activate GradeCommand
+return
+return
+return
+
+destroy GradeParser
+
+[->GradeCommand:execute(moduleList, configuration)
+activate GradeCommand
+GradeCommand -> ModuleList: getModule(moduleCode)
+activate ModuleList
+return
+
```
Perhaps might be worth considering leaving out the arguments for the methods as these are already being described in the text above and the diagrams are to essentially complement the explanation.

striris reviewed (commented):
This is a clear and impressive DG. Can tell that developers have spent a lot of time writing it! good job :)

hearobe commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+
+The following sequence diagram illustrates the process:
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/EditSeqDiagrams/Edit.puml)
```
![image](https://user-images.githubusercontent.com/65173208/161183852-e26cadfd-0c84-4974-a6ed-039687eae75e.png)

Should there be a :Main class that calls the parseCommand and execute methods? It might be hard for viewers where these messages originate from otherwise

OzairHasan commented on docs/SequenceDiagrams/EditSeqDiagrams/Edit.puml:
```diff
@@ -0,0 +1,52 @@
+
+note right of ModHappyParser
+Some methods are omitted from this diagram.
+end note
+
+[->ModHappyParser:parseCommand(userInput)
+activate ModHappyParser
+create EditParser
+ModHappyParser -> EditParser: EditParser()
+activate EditParser
+return
+
+ModHappyParser -> EditParser: parseCommand(arguments)
+activate EditParser
+create EditCommand
+EditParser -> EditCommand: EditCommand(taskModule,taskIndex,description,workingTime,taskName)
+activate EditCommand
+return
+return
+return
+
+destroy EditParser
+
+[->EditCommand:execute(moduleList, configuration)
+activate EditCommand
+ref over EditCommand, ModuleList: Get Module
+EditCommand -> EditCommand: editTaskFromModule(targetModule)
+activate EditCommand
+ref over EditCommand, TaskList: Get Task
+EditCommand -> Task: setTaskDescription(description)
+activate Task
+return
+deactivate Task
+return CommandResult(result)
+deactivate EditCommand
+
+destroy EditCommand
+
+
+@enduml
```
As mentioned for grade.puml, if the method arguments were left out here as well, the diagram would be much simpler/concise and easier to read.

hearobe commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/EditSeqDiagrams/Edit.puml)
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/EditSeqDiagrams/GetModule.puml)
```
![image](https://user-images.githubusercontent.com/65173208/161184194-d92ff3bb-42b3-457f-ad57-ab1f708da324.png)

Really like that you put some methods into a separate ref frame! It makes the diagram a lot clearer. For the condition in the alt frames, I wonder if it would be better to say something like [taskmodule == null] instead? It might be shorter and more readable that way.

OzairHasan commented on docs/SequenceDiagrams/EditSeqDiagrams/GetModule.puml:
```diff
@@ -0,0 +1,34 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+
+skinparam shadowing false
+participant ":EditCommand" as EditCommand
+participant ":ModuleList" as ModuleList
+hide footbox
+
+mainframe **sd** Get Module
+
+activate EditCommand
+EditCommand -> EditCommand: getTargetModule(moduleList)
+activate EditCommand
+alt Objects.isNull(taskModule)
+    EditCommand -> ModuleList: getGeneralTasks()
+    activate ModuleList
+    return
+
+else else
+    EditCommand -> ModuleList: getModule(taskModule)
+    activate ModuleList
+
+    alt Objects.isNull(taskModule)
+        [<-- ModuleList: throw NoSuchModuleException
+
+    else else
+    return targetModule
+
+    end
+end
+return
+deactivate EditCommand
+
+@enduml
```
Perhaps simplifying this diagram (there are 2 alt frames with the same condition and not sure the else condition belongs to which alt frame) and making it clearer would help to aid understanding

hearobe commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+
+Below is the sequence diagram of how the Grade feature works:
+
+![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/SequenceDiagrams/Grade.puml)
```
![image](https://user-images.githubusercontent.com/65173208/161184608-ef5a489a-df9a-4c57-9c87-0bf518280550.png)

I'm not too sure about this, but should the X for end of lifeline be below the return statement rather than on the statement itself?

OzairHasan reviewed (commented):
Overall a very good effort with most of the diagrams and The DG looks good as well.

YitHien reviewed (commented):
A great DG overall! Really shows a lot of effort your team has put in for this project!

hearobe commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,337 @@
+* `Command`: Handles command execution logic.
+* `Data`: Manages module and task data in program memory.
+* `Storage`: Reads data from, and writes data to Mod Happy's data storage files.
+
```
This is a really clear explanation and diagram of the entire design of the program that's easily understood!

hearobe reviewed (commented):
A good Developer Guide overall :)

Yzkkk committed:
made some error messages more informative

Yzkkk committed:
made changes on documentation

Yzkkk committed:
minor edits

Yzkkk committed:
minor edits

Yzkkk committed:
commented some of the test cases

Yzkkk committed:
commented junit test case

Yzkkk committed:
added exception to account for spaces added as parameters

Yzkkk committed:
minor edits

Yzkkk committed:
minor edits

Yzkkk committed:
minor edits

Yzkkk committed:
minor edits

ngys117 committed:
Fix unhandled null exception when deleting a module that does
not exist.

Fix unhandled out of bounds exception when editing tasks.

ngys117 committed:
Refactor code to reduce arrowhead

Remove unecessary exception check from removeModule as it is handled
by deleteModule in DeleteCommand.

Yzkkk committed:
made error messages and exceptions more descriptive

Yzkkk committed:
minor edits

Yzkkk committed:
commented out some test cases

ngys117 committed:
Fix Typos

ngys117 committed:
Remove number of tasks message

ngys117 committed:
Add JUnit test

Refactor some code

Ch40gRv1-Mu committed:
Update UG

Add expected output to examples.

ngys117 committed:
Fix editing a non-existent module throwing a null exception

Ch40gRv1-Mu committed:
Updates user guide

Add example input in command summary

ngys117 committed:
Add comment to DeleteCommand

Ch40gRv1-Mu committed:
Update Developer guide

Add application of Gson

Ch40gRv1-Mu committed:
Merge pull request #171 from ngys117/branch-PR-BugFixes

Ui Bug fixes and some code refactoring. Put the comments of heekit as a new issue, which will be fixed in future PR.

Ch40gRv1-Mu committed:
Merge branch 'master' into branch-update-ug-dg-fix-bugs

ngys117 committed:
Refactor Code for Issue #172

Ch40gRv1-Mu committed:
Update User guide

Fix a confusing discription.

Ch40gRv1-Mu committed:
Merge branch 'branch-update-ug-dg-fix-bugs' of https://github.com/Ch40gRv1-Mu/tp into branch-update-ug-dg-fix-bugs

Ch40gRv1-Mu committed:
Upgrade UG

Upgrade UG

Ch40gRv1-Mu committed:
Update UG

Update UG

heekit73098 committed:
Merge pull request #173 from Ch40gRv1-Mu/branch-update-ug-dg-fix-bugs

Branch update ug dg fix bugs

Yzkkk committed:
documented regex

Yzkkk committed:
minor edits

chooyikai committed:
Code cleanup as per TA feedback

chooyikai committed:
Fix #136 and #140

Yzkkk committed:
fixed bugs

Yzkkk committed:
Merge branch 'master' into master

Yzkkk committed:
minor edits

Yzkkk committed:
resolved merge conflict

Yzkkk committed:
minor edits

Yzkkk committed:
Merge branch 'master' of https://github.com/Yzkkk/tp

# Conflicts:
#	src/test/java/seedu/duke/parsers/ModHappyParserTest.java

chooyikai committed:
Attempt to fix UG formatting

chooyikai committed:
Fix some inconsistencies

chooyikai committed:
Update user input prompt to match UG

chooyikai committed:
Add PPP draft

Yzkkk committed:
updated code quality

ngys117 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp

ngys117 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp into branch-PR-BugFixes

heekit73098 committed:
Merge pull request #123 from Yzkkk/master

added javadoc and fixed some bugs

ngys117 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp into branch-PR-BugFixes

ngys117 committed:
Add Product Portfolio Page

Rename an exception

chooyikai committed:
Merge branch 'master' into bugfix

ngys117 committed:
Update tag op exception

ngys117 committed:
Add InvalidFlagException to Junit

ngys117 committed:
Update ModuleList method according to suggestion

Update relevant JavaDoc

ngys117 committed:
Update Junit

ngys117 committed:
Fix PPP headings

ngys117 committed:
Update AboutUs.md

ngys117 committed:
Update AboutUs.md

Fix Portfolio Link

chooyikai committed:
Fix hyperlink error in about us

chooyikai committed:
Merge pull request #174 from ngys117/branch-PR-BugFixes

Refactor Code for Issue #172, Add PPP

chooyikai committed:
Merge branch 'master' into bugfix

ngys117 committed:
Merge pull request #175 from chooyikai/bugfix

Code cleanup + fix some bugs

heekit73098 committed:
Added more descriptive error messages

heekit73098 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp into branch-bugfix

heekit73098 committed:
Update TagException

heekit73098 committed:
Update code quality

Ch40gRv1-Mu committed:
Add PPP of Mu Changrui

Add PPP of Mu Changrui

heekit73098 committed:
update PPP

heekit73098 committed:
Update code based on comments

heekit73098 committed:
update code quality

Yzkkk committed:
Merge pull request #176 from heekit73098/branch-bugfix

Updated regex for more descriptive error messages

heekit73098 committed:
Merge pull request #178 from heekit73098/branch-PPP

Added PPP

Ch40gRv1-Mu committed:
Merge pull request #1 from Ch40gRv1-Mu/master

Pull updated upstream

Ch40gRv1-Mu committed:
Update Task

Fix issue [#154](https://github.com/AY2122S2-CS2113T-T10-3/tp/issues/154)

Ch40gRv1-Mu committed:
Update Main, Add ModHappyStorageManager

Add  singleton class ModHappyStorageManager that manage the load and write data in ModHappy

Ch40gRv1-Mu committed:
Update the ModHappyStorageManager

Shift the module validation from ModuleListStorage to ModHappyStorageManager to keep the logic of ModHappyStorageManager clear and simple.

Ch40gRv1-Mu committed:
Update the TaskDuration

Add "Mins" and "mins"

Ch40gRv1-Mu committed:
Update the ModHappyParserTest

Update the ModHappyParserTest

Yzkkk committed:
Edited task duration format and added PPP

Ch40gRv1-Mu committed:
Update Userguid

Keep the user guide updated, keep the indentation consistent.

Yzkkk committed:
updated UG

Yzkkk committed:
updated PPP

Ch40gRv1-Mu committed:
Modify name of PPP

Modify name of PPP

Ch40gRv1-Mu committed:
Add Changrui's PPP

Add Changrui's PPP

heekit73098 committed:
Merge pull request #181 from Ch40gRv1-Mu/branch-A-PPP

Add Changrui's PPP

Yzkkk committed:
updated PPP

Yzkkk committed:
Merge branch 'master' into master

Yzkkk committed:
updated PPP

Yzkkk committed:
Merge branch 'master' of https://github.com/Yzkkk/tp

Ch40gRv1-Mu committed:
Merge pull request #179 from Yzkkk/master

Edited task duration format and added PPP

heekit73098 committed:
Update DG

heekit73098 committed:
Update formatting

Yzkkk committed:
fixed bug

Yzkkk committed:
updated code quality

Ch40gRv1-Mu committed:
Update PPP

Update PPP

Ch40gRv1-Mu committed:
Update PPP and ModHappyStorageManager

Update PPP and ModHappyStorageManager

Ch40gRv1-Mu committed:
Fix coding standard

Fix coding standard

Ch40gRv1-Mu committed:
Merge branch 'master' into branch-add-ppp-fix-bug-refactoring-storage

Ch40gRv1-Mu committed:
Update ModHappyStorageManager

Update ModHappyStorageManager

chooyikai committed:
Attempt to fix github pages formatting errors

chooyikai committed:
Merge pull request #184 from chooyikai/ug-bugfix-attempt

Attempt to fix github pages formatting errors

chooyikai committed:
Attempt to fix github pages formatting issues... again

chooyikai committed:
Merge pull request #185 from chooyikai/ug-bugfix-attempt

Attempt to fix github pages formatting issues... again

Ch40gRv1-Mu committed:
Merge branch 'master' into branch-add-ppp-fix-bug-refactoring-storage

Ch40gRv1-Mu committed:
Update Userguide

Update Userguide

Ch40gRv1-Mu committed:
Update Userguide

Update Userguide

Ch40gRv1-Mu committed:
Update Userguide.md

Update Userguide.md

chooyikai committed:
Merge pull request #180 from Ch40gRv1-Mu/branch-add-ppp-fix-bug-refactoring-storage

Add ppp; fix bug#154, #134, #135; refactoring storage;

heekit73098 committed:
Update based on comments and bug fixes

heekit73098 committed:
update code quality

heekit73098 committed:
Added javadoc and more parser bug fix

chooyikai committed:
Merge pull request #182 from heekit73098/branch-PPP

Updated DG with instructions for manual testing and diagrams

Ch40gRv1-Mu committed:
Update UG and puml

Update UG and puml:
1. Synchronize the description of storage and corresponding diagram to be consistent with the updated code.
2. Update the EditParser diagram.

Ch40gRv1-Mu committed:
Update PPP

Update PPP

Ch40gRv1-Mu committed:
Update DG

Update DG

Ch40gRv1-Mu committed:
Update the exception catch and deleted unremoved system out

Update the exception catch and deleted unremoved system out

Ch40gRv1-Mu committed:
Remove unused import

Remove unused import

Ch40gRv1-Mu committed:
Improve code quality

Improve code quality

Ch40gRv1-Mu committed:
Updates Storage.puml

Synchronize to the updated implementation.

heekit73098 committed:
Basic refactoring for parsers for code quality

heekit73098 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp into branch-PPP

heekit73098 committed:
update ppp

chooyikai committed:
Merge pull request #186 from Ch40gRv1-Mu/branch-update-dg

update dg, improve code quality

Ch40gRv1-Mu committed:
Merge pull request #187 from heekit73098/branch-PPP

Small refactoring for parsers, no change to existing implementation

Ch40gRv1-Mu committed:
Improve Coding Quality

Improve Coding Quality

chooyikai committed:
Merge pull request #188 from Ch40gRv1-Mu/branch-improve-coding-quality

Improve Coding Quality

ngys117 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp

heekit73098 committed:
Added authorship

ngys117 committed:
Merge pull request #190 from heekit73098/branch-authorship

Added authorship

ngys117 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp

ngys117 committed:
Add author tags

Ch40gRv1-Mu committed:
Merge pull request #191 from ngys117/master

Add author tags

Ch40gRv1-Mu committed:
Declare Authorship and fix #189

Declare authorship and fix issue [#189](https://github.com/AY2122S2-CS2113T-T10-3/tp/issues/189)

Ch40gRv1-Mu committed:
Fix coding standard

Fix coding standard

Ch40gRv1-Mu committed:
Refactoring the authorship tag

Refactoring the authorship tag

Ch40gRv1-Mu committed:
Update contribution for Ui

Update contribution for Ui

chooyikai committed:
Merge pull request #192 from Ch40gRv1-Mu/branch-authorship-fix-bug

Add authorship fix issue [#189](https://github.com/AY2122S2-CS2113T-T10-3/tp/issues/189)

chooyikai committed:
Add reposense authorship tags

chooyikai committed:
Improve wording of some error messages

ngys117 committed:
Merge pull request #193 from chooyikai/reposense

Add reposense authorship tags

heekit73098 committed:
Update code quality and minor bug fixes

heekit73098 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp into branch-authorship

chooyikai committed:
Add data verification for config file

chooyikai committed:
Add more checks for task and module loading

chooyikai committed:
Fix for null grade

chooyikai committed:
Add reposense tags to new files

chooyikai committed:
Update DG with details on data-load error handling

heekit73098 committed:
Added Javadoc, bugfix, and allowed "" for editing a task

heekit73098 committed:
Update code quality

heekit73098 committed:
update pr based on comments

heekit73098 committed:
Update PPP

chooyikai committed:
Improve code quality

Ch40gRv1-Mu committed:
Merge pull request #196 from heekit73098/branch-authorship

Added JavaDoc, bug fixes, and  allowing "" for editing a task parameter

chooyikai committed:
Merge branch 'master' into config-bugfix

chooyikai committed:
Fix error when removing task description

Yzkkk committed:
resolved merge conflict

chooyikai committed:
Update UG to explain removing parameters

chooyikai committed:
Fix typo in DG

Yzkkk committed:
resolved merge conflict, added authorship

heekit73098 committed:
Merge pull request #194 from chooyikai/config-bugfix

Fix issues relating to loading of invalid data from file

Yzkkk committed:
minor edits

Yzkkk committed:
minor edits

Yzkkk committed:
minor edits

Yzkkk committed:
no message

heekit73098 committed:
Added support for module grade removal

heekit73098 committed:
Update PPP

heekit73098 committed:
fixed code quality

heekit73098 committed:
Fix code quality

heekit73098 committed:
Merge pull request #183 from Yzkkk/master

minor fixing, added authorship

heekit73098 committed:
Merge branch 'master' of https://github.com/AY2122S2-CS2113T-T10-3/tp into branch-authorship

chooyikai committed:
Merge pull request #197 from heekit73098/branch-authorship

Added support for module grade removal

Yzkkk committed:
minor fixing ad added javadoc for exceptions

Yzkkk committed:
updated PPP

Yzkkk committed:
minor fixing

chooyikai committed:
Fix UG/DG typos and consistency issues

chooyikai committed:
Merge pull request #198 from Yzkkk/master

standardizations and added javadoc for exceptions

chooyikai committed:
Merge branch 'master' into ugdg-typos

chooyikai committed:
Fix merge issues

chooyikai committed:
Update UG and PPP, fix bug in config loading

heekit73098 committed:
Merge pull request #199 from chooyikai/ugdg-typos

Fix UG/DG typos and consistency issues

chooyikai committed:
Fix broken table for the nth time

chooyikai committed:
Minor fix to DG

chooyikai committed:
Update DG testing instructions

heekit73098 committed:
Merge pull request #200 from chooyikai/table-fix

DG fixes

Ch40gRv1-Mu committed:
Update ch40grv1-mu.md

Update ch40grv1-mu.md

Ch40gRv1-Mu committed:
Updated ch40grv1-mu.md

Updated ch40grv1-mu.md

ngys117 committed:
Merge pull request #201 from Ch40gRv1-Mu/branch-update-PPP

Update ch40grv1-mu.md
